### PR TITLE
HLRC: Add ML get categories API

### PR DESCRIPTION
**Pull request by edsavage**
Relates #29827

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java [hunk 1]
```diff
@@ -426,6 +428,44 @@ public void getBucketsAsync(GetBucketsRequest request, RequestOptions options, A
...
 
+    /**
+     * Gets the categories for a Machine Learning Job.
+     * <p>
+     * For additional info
+     * see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-get-category.html">
+     *     ML GET categories documentation</a>
+     *
+     * @param request  The request
+     * @param options  Additional request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java]: I know some methods add the `@throws` and others don't. I think we should still add the `@throws` entry even if it is a checked exception.

I have not been consistent about this myself either :(.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetCategoriesRequest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+/**
+ * A request to retrieve categories of a given job
+ */
+public class GetCategoriesRequest extends ActionRequest implements ToXContentObject {
+
+
+    public static final ParseField CATEGORY_ID = new ParseField("category_id");
+
+    public static final ObjectParser<GetCategoriesRequest, Void> PARSER =
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetCategoriesRequest.java]: This should be `ConstructingObjectParser`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetCategoriesRequest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        new ObjectParser<>("get_categories_request", GetCategoriesRequest::new);
+
+
+    static {
+        PARSER.declareString((GetCategoriesRequest, jobId) -> GetCategoriesRequest.jobId = jobId, Job.ID);
+        PARSER.declareLong(GetCategoriesRequest::setCategoryId, CATEGORY_ID);
+        PARSER.declareObject(GetCategoriesRequest::setPageParams, PageParams.PARSER, PageParams.PAGE);
+    }
+
+    private String jobId;
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetCategoriesRequest.java]: This should be `final`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetCategoriesRequest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    private Long categoryId;
+    private PageParams pageParams;
+
+    private GetCategoriesRequest() {}
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/GetCategoriesRequest.java]: We should not have this ctr, and just use the `ConstructingObjectParser` to pass the `jobId` as it is required.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java (hunk 1 above)
Comment (benwtrent) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java]: @dimitris-athanasiou may have a different opinion around this ^, So I will defer to him.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningGetResultsIT.java [hunk 5]
```diff
@@ -90,6 +92,9 @@ public void createJobAndIndexResults() throws IOException {
         addBucketIndexRequest(time, true, bulkRequest);
         addRecordIndexRequest(time, true, bulkRequest);
 
+        // index some category results
+        addCategoriesIndexRequests(bulkRequest);
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningGetResultsIT.java]: I think we could perform this just in the get categories test. However, I suppose it doesn't really add to the execution time. Note that this setup is run once per test. Might be fun trying to see if there's a noticeable difference. :-)

Comment (edsavage) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningGetResultsIT.java]: Good point! That call is only indexing 3 category results so I suspect the impact on execution time is low but I'll make the change regardless. Anything we can do to speed up the tests has to be a good thing...

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java (hunk 1 above)
Comment (dimitris-athanasiou) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java]: You're right @benwtrent, we've been dropping the `@throws` clause in some of the methods in the client. We'll need to revisit and add them. I'll make a note to do that.

**Review by benwtrent — APPROVED** [review]

**Review by dimitris-athanasiou — COMMENTED** [review]
LGTM

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM2

**Pull request merged**